core/src/main/scala/sudoku/sudoku.scala

```diff
@@ -123,14 +123,12 @@ case class Sudoku(grid: Vector[Vector[Cell]]) {
 object Sudoku {
 
   def parse(s: String): Option[Sudoku] = {
-    sequence(s
-      .toVector
-      .map {
-        case '.' => Some(Undetermined((1 to 9).toList))
-        case w => Try(w.toString.toInt).toOption.map(Fixed)
-      }).map(grid => Sudoku(grid
-      .grouped(9)
-      .toVector))
+    val sudokuString = if (s.length == 9 * 9) Some(s) else None
```

Comment:
I wonder how could you make the following more readable. Maybe split into smaller function like `def validateSize(s: String): Option[Unit] = if (s.length == 9 * 9) Some(()) else None` and / or a for comprehension:
```
for {
  _      <- validateSize(s)
  grid <-  traverse(s.toVector)(validateCell)
} yield Sudoku(grid.grouped(9).toVector)
```